src/ripple/app/data/SociDB.cpp

```diff
@@ -172,112 +172,78 @@ namespace {
     is the default behavior of sqlite. We may be able to remove this
     class.
 */
-class WALCheckpointer : public Checkpointer, private beast::Thread
+class WALCheckpointer : public Checkpointer
 {
 public:
-    using Connection = sqlite_api::sqlite3;
+    WALCheckpointer (sqlite_api::sqlite3& conn, JobQueue& q)
+            : conn_ (conn), jobQueue_ (q)
+    {
+        sqlite_api::sqlite3_wal_hook (&conn_, &sqliteWALHook, this);
+    }
 
-    WALCheckpointer (sqlite_api::sqlite3&, JobQueue&);
-    ~WALCheckpointer () override;
-
-    static
-    int sqliteWALHook (void* s, sqlite_api::sqlite3*,
-                       const char* dbName, int walSize);
+    ~WALCheckpointer () override = default;
 
 private:
-    void runCheckpoint (const char* db, int walSize);
-    void run ();
-    void checkpoint ();
-
     using LockType = std::mutex;
```

Comment:
I misnamed this. It should be `MutexType`. Not part of this PR, but you could rename this while you're mucking with this file.